init/modules/EESSI/2023.06.lua

```diff
@@ -210,6 +210,22 @@ if os.getenv("EESSI_MODULE_STICKY") then
     load_message = load_message .. " (requires '--force' option to unload or purge)"
 end
 
+-- Filter system paths from LD_LIBRARY_PATH
+-- Needs to be reversible so first make a copy
+append_path ("EESSI_DEFAULT_HOST_LD_LIBRARY_PATH", os.getenv("LD_LIBRARY_PATH") or "")
+-- on unload the variable will no longer exist
+if mode() == "load" then
+    -- remove any standard paths that can interfere with the compat layer
+    remove_path ("EESSI_DEFAULT_HOST_LD_LIBRARY_PATH", "/lib")
+    remove_path ("EESSI_DEFAULT_HOST_LD_LIBRARY_PATH", "/lib64")
+    remove_path ("EESSI_DEFAULT_HOST_LD_LIBRARY_PATH", "/usr/lib")
+    remove_path ("EESSI_DEFAULT_HOST_LD_LIBRARY_PATH", "/usr/lib64")
+    remove_path ("EESSI_DEFAULT_HOST_LD_LIBRARY_PATH", "/usr/local/lib")
+    remove_path ("EESSI_DEFAULT_HOST_LD_LIBRARY_PATH", "/usr/local/lib64")
```

Comment:
I'm confused, didn't you intend to remove from `LD_LIBRARY_PATH` here? I.e. now, you are essentially leaving `LD_LIBRARY_PATH` unsanitized, while you're cleaning `EESSI_DEFAULT_HOST_LD_LIBRARY_PATH` (which I think you meant to keep as a the original value so you can restore later)

Reply:
I mean, I see that your test passes, I just don't understand how... I don't think it should pass with the current code...